waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -83,15 +88,22 @@ genServer spec =
     genFileCopy = return . C.mkTmplFd
 
 genDotEnv :: AppSpec -> Generator [FileDraft]
-genDotEnv spec = return $
-  case AS.dotEnvServerFile spec of
-    Just srcFilePath
-      | not $ AS.isBuild spec ->
-          [ createCopyFileDraft
-              (C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir)
-              srcFilePath
-          ]
-    _ -> []
+-- Don't generate .env if we are building for production, since .env is to be used only for
+-- development.
+genDotEnv spec | AS.isBuild spec = return []
+genDotEnv spec =
+  return
+    [ createTextFileDraft
+        (C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir)
+        (envVarsToDotEnvContent envVars)
+    ]
+  where
+    envVars = waspEnvVars ++ userEnvVars
+    userEnvVars = AS.devEnvVarsServer spec
+    waspEnvVars = case AS.devDatabaseUrl spec of
+      Just url | not isThereCustomDbUrl -> [(databaseUrlEnvVarName, url)]
+      _ -> []
+    isThereCustomDbUrl = any ((== databaseUrlEnvVarName) . fst) userEnvVars
```

Comment:
Nit: This block of code feels a bit messy, maybe extract it into a separate helper function

Reply:
`getWaspEnvVars` or similar

Reply:
I am up for making it clearer but I have to admit I am not sure how to go about it.
This is stuff in the context of dot env file -> we have all env vars, which are really wasp env vars or user env vars, and then we define how to get user env vars, how to get wasp env vars, and finally a check for if there is custom db url.

What do you find unclear, how would you suggest improving it? I think taking it out into standalone helper function would not be great because it would be missing the context of the server dot env file.